### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Uggbert** [IBM/mac-ibm-notifications#228]
**Describe the bug**
Setting the system time to 12hr allows datepicker to go beyond the set /end_date

**To Reproduce**

1. Set to 12hr time
2. run -type popup -accessory_view_type datepicker -accessory_view_payload "/title Test /start_date 2024/10/11 17:34:36 /end_date 2024/10/24 00:00:00"
3. able to select beyond /end_date
4. Set 24hr system time
5. run as above
6. unable to select beyond /end_date

**Expected behavior**
Regardless of 12hr or 24hr time, the user should not be able to select a date beyond the /end_date

**Screenshots**

**Desktop (please complete the following information):**
 - macOS 15
 - IBM Notifier v 3.2.1

**Additional context**
First noticed this issue when using SUPERMAN, but was able to re-create with just Notifier.

**Comment by Macjutsu** [comment]
I can confirm this bug... but... only with the combination of UK Region settings (in Language & Region) along with 12 hour time format (in Date & Time)... fun!

I'm not sure if other regions have this bug... but if the Region is set to USA then both 12 and 24 hour time formats work.

**Comment by Uggbert** [comment]
Can confirm, I am using UK regional settings. Should have mentioned that at
the time.

Kind Regards,
              Graeme Urquhart

**Comment by SMartorelli** [comment]
Yes I was able to replicate it, I just need to check if it's an Apple bug or an IBM Notifier one. Will update this thread with the answer!

**Comment by Uggbert** [comment]
@SMartorelli any update on this one?
